### DISCUSSION TITLE
Key is invalid. You must supply a key in OpenSSH public key format error message

### DISCUSSION THREAD

**Discussion by jjberg83** [community/community#23464] [New to GitHub]
<p>I´ve been working my way through the documentation for connecting with SSH key starting at:<br>
<a href="https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys" rel="noopener nofollow ugc">https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys</a></p>
<p>However, when I try to add my new SSH key I get the following error message:<br>
Key is invalid. You must supply a key in OpenSSH public key format</p>
<p>I´ve read a few earlier posts about this problem, and saw in one of them that the length should be 68 without the email. Mine is 50 characters long without the email. With the email included, it´s 67. The key starts with SHA256 and ends with my email address (with a lot of random signs in between.</p>
<p>I´m doing this on an iMac, and I few days ago I did it on my Macbook (and it went smoothly). However, in the meantime I´ve installed the newest version of MacOs on my iMac (Monterey, version 12.0.1). Could this somehow have screwed things up? Any tips would be fantastic!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="jjberg83" data-post="1" data-topic="209344">
<div class="title">
<div class="quote-controls"></div>
 jjberg83:</div>
<blockquote>
<p>The key starts with SHA256 and ends with my email address (with a lot of random signs in between.</p>
</blockquote>
</aside>
<p>That sounds like you’re trying to use the fingerprint instead of the public key. With OpenSSH the public key should be in a file with a name ending in <code>.pub</code>, and the content should start with <code>ssh-</code>, followed by the key type (e.g. <code>rsa</code> or <code>ed25519</code>).</p>

**Comment by jjberg83** [comment]
<p>Thanks a lot <a class="mention" href="/airtower-luna">@airtower-luna</a> ! Made my day :smile:</p>

**Comment by taxi100** [comment]
-----BEGIN CERTIFICATE-----
MIIGLDCCBRSgAwIBAgISA8CPq2LKu5rspdFV60P/VzHQMA0GCSqGSIb3DQEBCwUA
MDIxCzAJBgNVBAYTAlVTMRYwFAYDVQQKEw1MZXQncyBFbmNyeXB0MQswCQYDVQQD
EwJSMzAeFw0yMzA5MDIwOTU5MTVaFw0yMzEyMDEwOTU5MTRaMBQxEjAQBgNVBAMT
CTE4NTBjby5pcjCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBAM2qwB43
b4FhlbbzrwK5XjLxeBSCPmOlFy4kP1hN80ZrddWNVPO+xYAQGUtCJ1drAGwq9Dp4
oDlpwKlQqrMg8dstUL3tT6exenywedRXx1ht0vvq311e0pVWQ/0N8A6q06qZbPEX
CfBB45eiwXBlvIPFTCKPEz+NQuOAbpEtuQB++afVOmp5cBq5rL/NAb4R9yJyPKHAM4
ELi2AvMXlYGSk4n5ofh+pCi0M/Srjhj03clJ7Wm3pjrPI991hDDg9f7sXTWA/5E9
Gdm1umzAIPXRzxcTWXxhMD+TAcxP22+sBn5bih8LxGk7hxD8V9UGa0AyPRLe8lkO
R4ZihdrzRlHDBDBYcX6qQmuEPljFDcy7WAIyHfOBu2E8OYXJrhxQfn2AR79uD6gW
TIK0mFmcY14uQ+VfMh+lVW7nW0hFYnXnMdHFzQwHGbHc/RptCRuKdMqFvASBcpE3
HDtmRvnb59kVOGGmT3n5DAg4x9mFyTkhMG1SjELa+Aknde10DJaU8C9b8EI7qMqn
klgiQ+oIzxyywCsE8Yfq5ISM5/qVX5P9E0gT8+W+dY44BbyaHaxW3KuBmsyFp+xt
lB1x1UqtTRy9EmLOtIkj5FsB8KXBDPnstxYyPRpR4Kf6fCyHw3UpjW65TGjLWGGI
/vFlyk3yGdqJYMnFaWt9KCl6EZU7NHtEvykxAgMBAAGjggJYMIICVDAOBgNVHQ8B
Af8EBAMCBaAwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMCMAwGA1UdEwEB
/wQCMAAwHQYDVR0OBBYEFD/BH3qnDMG6hXnA5bzN/QPcisBlMB8GA1UdIwQYMBaA
FBQusxe3WFbLrlAJQOYfr52LFMLGMFUGCCsGAQUFBwEBBEkwRzAhBggrBgEFBQcw
AYYVaHR0cDovL3IzLm8ubGVuY3Iub3JnMCIGCCsGAQUFBzAChhZodHRwOi8vcjMu
aS5sZW5jci5vcmcvMGEGA1UdEQRaMFiCCTE4NTBjby5pcoINZnRwLjE4NTBjby5p
coIObWFpbC4xODUwY28uaXKCDXBvcC4xODUwY28uaXKCDnNtdHAuMTg1MGNvLmly
gg13d3cuMTg1MGNvLmlyMBMGA1UdIAQMMAowCAYGZ4EMAQIBMIIBBAYKKwYBBAHW
eQIEAgSB9QSB8gDwAHcAtz77JN+cTbp18jnFulj0bF38Qs96nzXEnh0JgSXttJkA
AAGKVYwKoQAABAMASDBGAiEAvXxE8eGpxg31AttkjtDZ3qePZtkk0ELGKDaq8LTc
yJ4CIQDYCPu+NosIrBecZDdtdzJlIggaCNRaWeYWFiXKMLo1OAB1AK33vvp8/xDI
i509nB4+GGq0Zyldz7EMJMqFhjTr3IKKAAABilWMC0EAAAQDAEYwRAIgDf8PThO2
9lGGFv3/MbS6Wtccapv/AQVOFhlT6/E2crdECIGpTcJzSBUbf1UB3LSpETDzq0eBO
/Rs8cLVyxwbUOn6+MA0GCSqGSIb3DQEBCwUAA4IBAQBOOP3gz0mE+BQwEr1YIjc6
BhC5tuPU/h6wR9khEBC14qf9xk/3lRxgssHvZ5BGrc5oGoXjYOm31aks1GFxwxKlq
VjyR44FAu6nu+Y566uKhTGHS701LUrdYdLvXlL9MkTAw3AvchlgSOCwsr/na2wqU
75pJEAJgRP0/GCRnbNqkbWCizhPLnHJtgKRoQCt4junzS6MS90a0ry+k4/kgKtN2
2csSsiE47n41RZY7OLl1kOODZT+NOZcn596aLgg+VYTptzrsTRZpsRnfRYmT1hBL
l1dzf73nejFc6mblHrQK3s+7Ifghe2ZXth1sCxXgOEfJi2BFFH/KQmv8K/9O564E
-----END CERTIFICATE-----

**Comment by taxi100** [comment]
SHA256:wcH95nkL7ZkeRURHpLuoThIBEIIKkYgf9etD18PIx40
